### ISSUE TITLE
how does train.py work?

### ISSUE THREAD

**Issue by gauravk97** [experiencor/basic-yolo-keras#104]
I have a doubt, which is how does the model train using train.py ?

Because in line 123 in train.py ( https://github.com/experiencor/basic-yolo-keras/blob/master/train.py#L123), while calling yolo.train, the labels aren't passed at all to the fitgenerator.

Also in https://github.com/experiencor/basic-yolo-keras/blob/master/frontend.py#L404 , while making BatchGenerator objects to be passed to model.fit_generator do not accept train and valid labels although in the documentattion of model.fit_generator it is stated that:

```
generator: A generator. The output of the generator must be either
a tuple (inputs, targets)
a tuple (inputs, targets, sample_weights). All arrays should contain the same number of samples. 

```

I am a newbie in this field, so I thought that maybe posting this doubt of mine as issue would be resolved. Thanks.

**Comment by experiencor** [comment]
@gauravk97 You will find fit_generator in ```frontend.py```. This find contains the definition of ```yolo.train```.

I don't understand your second question. The documentation of ```model.fit_generator``` inputs and targets, which are defined in my ```preprocessing.py```.

**Comment by gauravk97** [comment]
Yes, I found fit_generator in frontend.py but I was not able to find that annotations(or ground truth) had been passed to it anywhere. Only inputs images have been passed and not annotations. And annotations are necessary for the object to be able to learn.

In train.py, line123 yolo.train is called and even though earlier in the script at line 74, train_labels is defined using parse_annotation defined in preprocessing.py, it is not passed to the train function.

I am sorry for not being clear previously. But this is my question, without getting any sort of labels or annotations of the ground truth, how does train.py work?

**Comment by experiencor** [comment]
@gauravk97 The image and the corresponding annotation are passed at the same time. You can check out line 229 in preprocessing.py. x_batch contains images and y_batch contains the annotations. The generator instantiated at line 404 of frontend.py generates both images and annotations.
